configs/components/openssl-fips.rb

```diff
@@ -0,0 +1,56 @@
+component "openssl-fips" do |pkg, settings, platform|
+
+  pkg.build_requires "puppet-runtime"
+
+  openssl_fips_details = JSON.parse(File.read('configs/components/openssl-fips.json'))
+  openssl_fips_version = openssl_fips_details['version']
+  openssl_fips_location = openssl_fips_details['location']
+
+  tarball_name = "#{pkg.get_name}-#{openssl_fips_version}.#{platform.name}.tar.gz"
+  pkg.url File.join(openssl_fips_location, tarball_name)
+  pkg.sha1sum File.join(openssl_fips_location, "#{tarball_name}.sha1")
+  pkg.version openssl_fips_version
+  pkg.install_only true
+
+  pkg.add_source("file://resources/patches/openssl/openssl-fips.cnf.patch")
+
+  # Overlay openssl-fips onto puppet-runtime in /opt
+  #
+  # fipsmodule.cnf needs to be generated at install time on the host, so never
+  # include it in the package.
+  #
+  # openssl.cnf also needs to be updated at install time, so don't include it in
+  # the package. Instead ship `openssl-fips.cnf` and overwrite `openssl.cnf`
+  # with that version after the `fips.so` library is installed.
+  #
+  pkg.install do
+    [
+      "gunzip -c #{tarball_name} | #{platform.tar} --skip-old-files -C / -xf -",
+      "rm -f /opt/puppetlabs/puppet/ssl/fipsmodule.cnf",
```

Comment:
I would be tempted to not do this or to remove the `-f`, assuming the preremove_action suggestion, below. If something is not as expected, I rather see an error than it to silently self-correct. 

This has the potential of unwanted annoyance through corner case(s) but I think I'd start with it.

Reply:
So this `--force` is because when building the package, I want to ensure the `fipsmodule.cnf` that the `openssl-fips` tarball laid down overlaid does not "leak" into the puppet-agent package. But if in the future `fipsmodule.cnf` is removed from the `openssl-fips` tarball, then it's ok to proceed.

Reply:
Removed `fipsmodule.cnf` in https://github.com/puppetlabs/openssl-fips/commit/6e94106d4e0796d881e294ca494c2c0a6d678ddf and removed [`rm`](https://github.com/puppetlabs/puppet-agent/compare/63787dfb2beb416732e52ec0903bab077ebf91d4..91df5ad0234e67560c5df10c3ab3045daad03cc9) command